sdk/go/src/sawtooth_sdk/client/signer.go

```diff
@@ -0,0 +1,106 @@
+package client
+
+import (
+	"crypto/sha256"
+	"crypto/sha512"
+	ellcurv "github.com/btcsuite/btcd/btcec"
+	"math/big"
+)
+
+// -- Key Gen --
+
+// GenPrivKey generates a new private key and returns it as bytes
+func GenPrivKey() []byte {
+	priv, err := ellcurv.NewPrivateKey(ellcurv.S256())
+	if err != nil {
+		panic("Failed to generate private key")
+	}
+	return priv.Serialize()
+}
+
+// GenPubKey generates a new public key from a given private key and returns it
+// using the 33 byte compressed format
+func GenPubKey(privkey []byte) []byte {
+	_, pub := ellcurv.PrivKeyFromBytes(ellcurv.S256(), privkey)
+	return pub.SerializeCompressed()
```

Comment:
My thought here was that it would be a little faster to pass around the byte array directly rather than the string encoded key (which would need to be encoded/decoded constantly). My understanding is there will be a separate guide for using the Go SDK which would give examples where you are encoding the key.